Sources/_NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -543,21 +543,32 @@ extension SystemFileHandle.SendableView {
                 named: name
             ).flatMapError { errno -> Result<[UInt8], FileSystemError> in
                 switch errno {
-                #if canImport(Darwin)
+                #if canImport(Darwin) || os(FreeBSD)
                 case .attributeNotFound:
                     // Okay, return empty value.
                     return .success([])
                 #endif
+                #if !os(FreeBSD)
                 case .noData:
                     // Okay, return empty value.
                     return .success([])
+                #endif
                 default:
+                    #if os(FreeBSD)
+                    let error = FileSystemError.extattr_get_fd(
```

Comment:
Does this behave meaningfully differently than `fgetxattr`? If not, can this `#if os` be moved into that function so it doesn't end up here?